components/org.wso2.carbon.editor.log.appender/src/main/java/org/wso2/carbon/editor/log/appender/EditorConsoleAppender.java

```diff
@@ -0,0 +1,132 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wso2.carbon.editor.log.appender;
+
+import org.apache.logging.log4j.core.Appender;
+import org.apache.logging.log4j.core.Core;
+import org.apache.logging.log4j.core.Filter;
+import org.apache.logging.log4j.core.Layout;
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.core.appender.AbstractAppender;
+import org.apache.logging.log4j.core.config.plugins.Plugin;
+import org.apache.logging.log4j.core.config.plugins.PluginAttribute;
+import org.apache.logging.log4j.core.config.plugins.PluginElement;
+import org.apache.logging.log4j.core.config.plugins.PluginFactory;
+import org.apache.logging.log4j.core.layout.PatternLayout;
+import org.apache.logging.log4j.core.util.Booleans;
+import org.wso2.carbon.editor.log.appender.internal.CircularBuffer;
+import org.wso2.carbon.editor.log.appender.internal.ConsoleLogEvent;
+
+import java.io.Serializable;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+/**
+ * This appender will be used to capture the logs and later send to clients, if requested via the
+ * logging web service.
+ * This maintains a circular buffer, of some fixed amount (say 100).
+ */
+@Plugin(name = "EditorConsole", category = Core.CATEGORY_NAME, elementType = Appender.ELEMENT_TYPE, printObject = true)
+public final class EditorConsoleAppender extends AbstractAppender {
+
+    private CircularBuffer<ConsoleLogEvent> circularBuffer;
+    private static final int BUFFER_SIZE = 1000;
+
+    /**
+     * Creates an instance of EditorConsoleAppender.
+     *
+     * @param name             appender name
+     * @param filter           null if not specified
+     * @param layout           pattern of log messages
+     * @param ignoreExceptions default is true
+     *                         <p>
+     *                         Called by {@link #createAppender(String, Filter, Layout, String, String)}
+     */
+    private EditorConsoleAppender(final String name, final int bufferSize, final Filter filter,
+                                  final Layout<? extends Serializable> layout, final boolean ignoreExceptions) {
+        super(name, filter, layout, ignoreExceptions);
+
+        activateOptions();
+    }
+
+    /**
+     * Taken from the previous EditorConsoleAppender
+     */
+    public void activateOptions() {
+        this.circularBuffer = DataHodlder.getBuffer(BUFFER_SIZE);
+    }
+
+    /**
+     * Creates a EditorConsoleAppender instance with
+     * attributes configured in log4j2.properties.
+     *
+     * @param name   appender name
+     * @param filter null if not specified
+     * @param layout pattern of log messages
+     * @param ignore default is true
+     * @return intance of EditorConsoleAppender
+     */
+    @PluginFactory
+    public static EditorConsoleAppender createAppender(@PluginAttribute("name") final String name,
```

Comment:
Check the java docs every where . missing parameters

Reply:
The Java doc comments are available. Please verify what's meant by Java docs missing parameters